### PR TITLE
Feature/make payment

### PR DESCRIPTION
**Pull request by ivanbelasich**
# Summary

This pull request introduces a new feature enabling users to generate a popup with instructions for making a payment. The popup includes details such as the recipient's Stellar public key, the amount to be paid, the type of asset (including non-native assets), and, if applicable, the public key of the asset issuer.

# Details

- Improved UX/UI design.
- Testing for functionality.
- Error handling implemented.
- Documentation provided: test.html and README.md.

# Evidence

image
image

image
image

### CODE REVIEW

File: cypress/integration/ui/payment_spec.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/// <reference types="cypress" />
+/// <reference types="@testing-library/cypress"/>
+import { amountToSend, assetType, destinationAccount, issuer } from '../../fixtures/payment.json';
+
+describe('checks that the /payment component works', () => {
+    const BASE_URL = '/payment';
+
+    it('should visit /sign with payment information but user is not connected', () => {
```
Comment (Ja-boop) [inline cypress/integration/ui/payment_spec.ts]: Shouldn't be visit /payment?

File: src/lib/bridge/Bridge.ts [hunk 2]
```diff
@@ -1,21 +1,25 @@
...
 export type IAvailableWalletsMessageHandler = (message: IAvailableWalletsMessage) => void;
 export type ITransactionMessageHandler = (message: ITransactionMessage) => void;
+export type IPaymentMessageHandler = (message: IPaymentMessage) => void;
 
 export enum SimpleSignerEventType {
     ON_CONNECT = 'onConnect',
     ON_READY = 'onReady',
     ON_SIGN = 'onSign',
     ON_CANCEL = 'onCancel',
+    ON_PAYMENT_REQUEST = 'onPayment',
```
Comment (Ja-boop) [inline src/lib/bridge/Bridge.ts]:
```suggestion
    ON_PAYMENT = 'onPayment',
```

File: src/lib/bridge/Bridge.ts [hunk 3]
```diff
@@ -106,4 +145,9 @@ export default class Bridge {
             window.opener.postMessage(message, '*');
         }
     }
+
+    public sendPaymentRequest(receiver: string, amount: number, assetType: string, issuer: string) {
```
Comment (Ja-boop) [inline src/lib/bridge/Bridge.ts]: I think this function has too many parameters, can you group them into a type or interface?

File: src/lib/stellar/Payment.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    amount: string,
+    assetType: string,
+    issuer?: string,
+) {
+    let asset;
+    if (assetType === 'native') {
+        asset = Asset.native();
+    } else {
+        asset = new Asset(assetType, issuer);
+    }
```
Comment (Ja-boop) [inline src/lib/stellar/Payment.ts]:
```suggestion
    const asset = assetType === 'native' ? Asset.native() : new Asset(assetType, issuer);
```

File: README.md [hunk 5]
```diff
@@ -106,7 +106,7 @@ you want to contribute.
 
 # How to implement Simple Signer on your website
 
-Simple Signer provides two endpoints, `/connect` and `/sign` which allow some customisations to be made.
+Simple Signer provides three endpoints, `/connect`, `/sign` and `/payment` which allow some customisations to be made.
```
Comment (Ja-boop) [inline README.md:109]:
```suggestion
Simple Signer provides three endpoints, `/connect`, `/sign` and `/payment` which allow some customizations to be made.
```

**Review by GonzaloAlt — APPROVED** [review]

**Pull request merged**